swarmform/core/clustering_algo/hrab_clustering.py

```diff
@@ -0,0 +1,233 @@
+from swarmform.core.swarm_dag import Node, DAG
+from swarmform import SwarmPad
+from swarmform.core.swarmwork import SwarmFlow
+
+
+def get_tasks_at_level(workflow, level):
+    """
+    return the tasks at given level
+
+    Args:
+        workflow (DAG): DAG object of workflow
+        level (int): level which output tasks needed to be
+
+    Returns:
+        list(Node)
+    """
+    tasks = []
+    for task in workflow.get_nodes():
+        if workflow.get_nodes()[task].get_level() == level:
+            tasks.append(workflow.get_nodes()[task])
+    return tasks
+
+
+def sort_tasks_by_exec_time_in_decreasing(tasks):
+    """
+    Sort tasks in descending order by the execution time
+
+    Args:
+        tasks (list(Node)
+
+    Returns:
+        list(Node)
+    """
+    n = len(tasks)
+    for i in range(n):
+        for j in range(0, n - i - 1):
+            if tasks[j].get_exec_time() < tasks[j + 1].get_exec_time():
+                tasks[j], tasks[j + 1] = tasks[j + 1], tasks[j]
+    return tasks
+
+
+def get_sum_0f_exec_time(cluster):
+    """
+    Returns the sum of execution time from a given list of tasks
+
+    Args:
+        cluster (list(Node))
+
+    Returns:
+        int
+    """
+    exec_sum = 0
+    for task in cluster:
+        exec_sum += task.get_exec_time()
+    return exec_sum
+
+
+def cluster_wf(workflow, cluster_num):
```

Comment:
suggestion: rename function name to 'rac_clustering' as same as 'wpa_clustering'

Reply:
+ add a docstring